backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/badge/BpkBadge.kt

```diff
@@ -60,26 +63,38 @@ fun BpkBadge(
     Row(
         modifier = modifier
             .semantics(mergeDescendants = true) { }
-            .requiredHeight(BpkSpacing.Lg)
             .border(1.dp, type.borderColor, BadgeShape)
             .background(type.backgroundColor, BadgeShape)
-            .padding(horizontal = BpkSpacing.Md),
+            .padding(horizontal = BpkSpacing.Md, vertical = BpkSpacing.Sm),
         horizontalArrangement = Arrangement.spacedBy(BpkSpacing.Sm),
         verticalAlignment = Alignment.CenterVertically,
     ) {
         val contentColor = type.contentColor
-        if (icon != null) {
+        val badgeIcon = when (type) {
+            BpkBadgeType.Warning -> BpkIcon.InformationCircle
+            BpkBadgeType.Destructive -> BpkIcon.Exclamation
+            BpkBadgeType.Success -> BpkIcon.TickCircle
+            else -> icon
+        }
+        if (badgeIcon != null) {
             BpkIcon(
-                icon = icon,
+                icon = badgeIcon,
                 contentDescription = null,
                 size = BpkIconSize.Small,
-                tint = contentColor,
+                tint = when (type) {
```

Comment:
nit: to make it a little more readable it might be better to create a variable for this (or even create an extension function, like `iconColor` in the same place the `contentColor` is defined to use this

Reply:
This should now be ok!